tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -188,7 +193,50 @@ shared_ptr<CKKSTensor> CKKSTensor::mul_plain_inplace(const double& to_mul) {
 }
 
 shared_ptr<CKKSTensor> CKKSTensor::sum_inplace(size_t axis) {
-    // TODO
+    if (axis >= _shape.size()) throw invalid_argument("invalid axis");
+
+    auto new_shape = _shape;
+    new_shape.erase(new_shape.begin() + axis);
+
+    if (_batch_size) {
+        if (axis != 0) throw invalid_argument("unsupported argument");
```

Comment:
Why isn't it possible to sum other axis when there is a batch_axis? It should be still possible to do so without affecting the batch_size, keeping it the same while summing up over the other axis